pkg/stream/batch_consumer.go

```diff
@@ -12,53 +11,101 @@ import (
 	"time"
 )
 
+const MetricNameConsumerReceivedCount = "ConsumerReceivedCount"
+const MetricNameConsumerProcessedCount = "ConsumerProcessedCount"
+
 //go:generate mockery -name=BatchConsumerCallback
 type BatchConsumerCallback interface {
-	Boot(config cfg.Config, logger mon.Logger)
-	Consume(ctx context.Context, msg []*Message) error
+	Boot(cfg.Config, mon.Logger) error
+	Consume(context.Context, []*Message) ([]*Message, error)
```

Comment:
@j4k4 suggested to keep parameter names in interfaces because they make it easier to know which parameter is for what.